x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -407,21 +415,37 @@ export class UninstallTokenService implements UninstallTokenServiceInterface {
 
   private async getPolicyIdsBatch(
     batchSize: number = SO_SEARCH_LIMIT,
-    page: number = 1
+    page: number = 1,
+    managedPoliciesOnly?: boolean
   ): Promise<string[]> {
-    return (
-      await agentPolicyService.list(this.soClient, { page, perPage: batchSize, fields: ['id'] })
-    ).items.map((policy) => policy.id);
+    if (managedPoliciesOnly) {
+      return (
+        await agentPolicyService.list(this.soClient, {
+          page,
+          perPage: batchSize,
+          fields: ['id', 'is_managed'],
```

Comment:
instead of filtering the `is_managed` in the response after, it would be more efficient to do it in the search via the [kuery param](https://github.com/elastic/kibana/blob/ac81d96f2ff9a59adce381d9bce11ba4b9a82bcf/x-pack/plugins/fleet/server/services/agent_policy.ts#L418).